src/containers/App/reducer.js

```diff
@@ -58,27 +74,38 @@ function appReducer(state = initialState, action) {
         .set('loading', false);
 
     case REQUEST_CATEGORIES_SUCCESS:
-      return state
-        .set('categories', fromJS(action.payload));
+      return state.set('categories', fromJS(action.payload));
 
     case UPLOAD_REQUEST:
-      return state
-        .set('upload', fromJS({
+      return state.set(
+        'upload',
+        fromJS({
           id: action.payload.id,
-          file: action.payload.file.name
-        }));
+          file: action.payload.file.name,
+        }),
+      );
 
     case UPLOAD_PROGRESS:
-      return state
-        .set('upload', fromJS({
+      return state.set(
+        'upload',
+        fromJS({
           ...state.get('upload').toJS(),
-          progress: action.payload
-        }));
+          progress: action.payload,
+        }),
+      );
 
     case UPLOAD_SUCCESS:
     case UPLOAD_FAILURE:
+      return state.set('upload', fromJS({}));
+
+    case LOGOUT:
+      global.sessionStorage.removeItem(ACCESS_TOKEN);
```

Comment:
de global.sessionStorage is being set in the saga already: the login and logout functions in auth.js

Reply:
Another good call; also removed it.